atoma-state/src/state_manager.rs

```diff
@@ -135,8 +136,38 @@ impl AtomaStateManager {
     /// ```
     #[instrument(level = "trace", skip_all)]
     pub async fn run(self, mut shutdown_signal: Receiver<bool>) -> Result<()> {
+        let mut network_metrics = NetworkMetrics::new();
+        let interval = std::time::Duration::from_secs(15);
+
+        // Create a channel for interval-based updates
+        let (interval_tx, interval_rx) = flume::unbounded();
+
+        // Spawn a task that sends a message every interval
+        tokio::spawn(async move {
```

Comment:
Why spawning a new task for this ? The atoma state manager is already spawned on a new task, in the main binary

Reply:
As mentioned [here](https://github.com/atoma-network/atoma-proxy/pull/444#discussion_r2068983119) having it in the same thread wouldn't work thus I created a separate thread. This shouldn't affect the other thread regardless so it should be fine